`cargo test` cannot pass due to csv-rs doesn't support openssl version:806354976

the issue can be found from here:
https://github.com/confidential-containers/guest-components/actions/runs/6105168876/job/16568311628?pr=321

```
Error: failed to run custom build command for `csv-rs v0.1.0 (https://gitee.com/anolis/csv-rs?rev=bcf3bcc#bcf3bcc0)`
Caused by:
  process didn't exit successfully: `/home/runner/work/guest-components/guest-components/target/debug/build/csv-rs-c4227a814abf3087/build-script-build` (exit status: 101)
  --- stderr
  thread 'main' panicked at 'Unsupported openssl version:806354976', /home/runner/.cargo/git/checkouts/csv-rs-6ce26198efa2f606/bcf3bcc/build.rs:19:18
```

and can be verified locally, 
```
cargo test --features openssl,rust-crypto,offline_fs_kbc -p attestation-agent -p attester -p coco_keyprovider -p kbc -p kbs_protocol -p attestation_agent -p crypto -p resource_uri -p sev@0.1.0
```

Tried to revert the change here: https://github.com/confidential-containers/guest-components/pull/342, but still doesn't work.

```
  = note: /usr/bin/ld: /go/src/github.com/confidential-containers/guest-components/target/debug/deps/libcsv_rs-cf8059eec6404d4e.rlib(csv_rs-cf8059eec6404d4e.csv_rs.fbdd6b72ef39f42a-cgu.11.rcgu.o): undefined reference to symbol 'EVP_PKEY_set_alias_type@@OPENSSL_1_1_1'
          /usr/bin/ld: /usr/lib/x86_64-linux-gnu/libcrypto.so.1.1: error adding symbols: DSO missing from command line
          collect2: error: ld returned 1 exit status
```

--- a/attestation-agent/attester/Cargo.toml
+++ b/attestation-agent/attester/Cargo.toml
@@ -19,7 +19,7 @@ sev = { version = "1.2.0", default-features = false, features = ["snp"], optiona
 strum.workspace = true
 tdx-attest-rs = { git = "https://github.com/intel/SGXDataCenterAttestationPrimitives", tag = "DCAP_1.16", optional = true }
 # TODO: change it to "0.1", once released.
-csv-rs = { git = "https://gitee.com/anolis/csv-rs", rev = "bcf3bcc", optional = true }
+csv-rs = { git = "https://gitee.com/anolis/csv-rs", rev = "05fbacd", optional = true }
 codicon = { version = "3.0", optional = true }
 hyper = { version = "0.14", features = ["full"], optional = true }
 hyper-tls = { version = "0.5", optional = true }

